site/src/job_queue.rs

```diff
@@ -36,6 +41,93 @@ async fn create_benchmark_request_master_commits(
     Ok(())
 }
 
+/// Parses strings in the following formats extracting the Date & release tag
+/// `static.rust-lang.org/dist/2016-05-24/channel-rust-1.9.0.toml`
+/// `static.rust-lang.org/dist/2016-05-31/channel-rust-nightly.toml`
+/// `static.rust-lang.org/dist/2016-06-01/channel-rust-beta.toml`
+/// `static.rust-lang.org/dist/2025-06-26/channel-rust-1.89-beta.toml`
+/// `static.rust-lang.org/dist/2025-06-26/channel-rust-1.89.0-beta.toml`
+/// `static.rust-lang.org/dist/2025-06-26/channel-rust-1.89.0-beta.2.toml`
+fn parse_release_string(url: &str) -> anyhow::Result<Option<(String, DateTime<Utc>)>> {
+    static VERSION_RE: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"(\d+\.\d+\.\d+)").unwrap());
+
+    // Grab ".../YYYY-MM-DD/FILE.toml" components with Path helpers.
+    let file = Path::new(url)
+        .file_name()
+        .and_then(|n| n.to_str())
+        .ok_or_else(|| anyhow::anyhow!("URL lacks a file name"))?;
+
+    let date_str = Path::new(url)
+        .parent()
+        .and_then(Path::file_name)
+        .and_then(|n| n.to_str())
+        .ok_or_else(|| anyhow::anyhow!("URL lacks a date segment"))?;
+
+    // No other beta releases are recognized as toolchains.
+    //
+    // We also have names like this:
+    //
+    // * channel-rust-1.75-beta.toml
+    // * channel-rust-1.75.0-beta.toml
+    // * channel-rust-1.75.0-beta.1.toml
+    //
+    // Which should get ignored for now, they're not consumable via rustup yet.
+    if file.contains("beta") && file != "channel-rust-beta.toml" {
+        return Ok(None);
+    }
+
+    // Parse the YYYY-MM-DD segment and stamp it with *current* UTC time.
+    let naive = NaiveDate::parse_from_str(date_str, "%Y-%m-%d")?;
+    let now = Utc::now();
```

Comment:
I don't think adding timestamps based on when was the artifact parsed is ideal, because then when we call the function multiple times, it will return different results. I would just assign a fixed time to all releases, e.g. the beginning of a day (`00:00`).